discovery/dns/dns.go

```diff
@@ -317,17 +317,19 @@ func askServerForName(name string, queryType uint16, client *dns.Client, servAdd
 	}
 
 	response, _, err := client.Exchange(msg, servAddr)
-	if err == dns.ErrTruncated {
+	if err != nil {
+		return nil, err
+	}
+
+	if response.Truncated {
 		if client.Net == "tcp" {
 			return nil, fmt.Errorf("got truncated message on TCP (64kiB limit exceeded?)")
 		}
 
 		client.Net = "tcp"
 		return askServerForName(name, queryType, client, servAddr, false)
 	}
-	if err != nil {
-		return nil, err
-	}
+
 	if msg.Id != response.Id {
```

Comment:
As per @simonpasquier , you can now remove this. This check is now redundant. See https://github.com/miekg/dns/blob/56be65265e34e731425e0269a301774938827c60/client.go#L432 for clarity on this.